src/climbing.cpp

```diff
@@ -0,0 +1,406 @@
+}
+
+void climbing_aid::condition::deserialize( const JsonObject &jo )
+{
+    mandatory( jo, false, "type", cat );
+    mandatory( jo, false, "flag", flag );
+    switch( cat ) {
+        case category::item: {
+            mandatory( jo, false, "uses", uses_item );
+            break;
+        }
+        case category::ter_furn: {
+            optional( jo, false, "range", range );
+            break;
+        }
+        default: {
+            break;
+        }
+    }
+
+}
+
+void climbing_aid::down_t::deserialize( const JsonObject &jo )
+{
+    // max_height is optional; setting it to zero disables "down" motion
+    optional( jo, true, "max_height", max_height );
+    optional( jo, true, "easy_climb_back_up", easy_climb_back_up );
+
+    if( enabled() ) {
+        // Mandatory for all aids that
+        mandatory( jo, was_loaded, "menu_text", menu_text );
+        mandatory( jo, was_loaded, "confirm_text", confirm_text );
+        std::string menu_hotkey_str;
+        if( deploys_furniture() ) {
+            mandatory( jo, was_loaded, "menu_cant", menu_cant );
+            mandatory( jo, was_loaded, "menu_hotkey", menu_hotkey_str );
+            if( menu_hotkey_str.length() != 1 ) {
+                jo.throw_error( str_cat( "failed to read mandatory member \"menu_hotkey\"" ) );
+            }
+        } else {
+            optional( jo, was_loaded, "menu_cant", menu_cant );
+            optional( jo, was_loaded, "menu_hotkey", menu_hotkey_str );
+            if( menu_hotkey_str.length() > 1 ) {
+                jo.throw_error( str_cat( "failed to read optional member \"menu_hotkey\"" ) );
+            }
+        }
+        if( menu_hotkey_str.length() ) {
+            menu_hotkey = menu_hotkey_str[ 0 ];
+        }
+
+        optional( jo, true, "msg_before", msg_before );
+        optional( jo, true, "msg_after", msg_after );
+        optional( jo, true, "cost", cost );
+        optional( jo, true, "deploy_furn", deploy_furn );
+    }
+
+    was_loaded = true;
+}
+
+void climbing_aid::climb_cost::deserialize( const JsonObject &jo )
+{
+    optional( jo, true, "kcal", kcal );
+    optional( jo, true, "thirst", thirst );
+    optional( jo, true, "damage", damage );
+    optional( jo, true, "pain", pain );
+}
+
+template< typename Lambda >
+void for_each_aid_condition( const climbing_aid::condition_list &conditions, const Lambda &func )
+{
+    for( auto &cond : conditions ) {
+        if( int( cond.cat ) < int( climbing_lookup.size() ) ) {
+            auto &cat = climbing_lookup[ int( cond.cat ) ];
+            auto range = cat.equal_range( cond.flag );
+            for( auto i = range.first; i != range.second; ++i ) {
+                func( cond, *i->second );
+            }
+        }
+    }
+}
+
+climbing_aid::aid_list climbing_aid::list(
+    const condition_list &conditions )
+{
+    std::vector<const climbing_aid *> list;
+
+    const auto add_deployables = [&]( const condition &, const climbing_aid & aid ) {
```

Comment:
I think capturing everything is bad style.
```suggestion
    const auto add_deployables = [&list]( const condition &, const climbing_aid & aid ) {
```